tests/execution_engine/test_sqlalchemy_execution_engine.py

```diff
@@ -1258,3 +1266,314 @@ def test__dialect_requires_persisted_connection_error_on_multiple_params_empty_u
+        condition = NullityCondition(column=Column(name="email"), is_null=False)
+        result = engine.condition_to_filter_clause(condition)
+        compiled = str(result.compile(compile_kwargs={"literal_binds": True}))
+        assert compiled == "email IS NOT NULL"
+
+    @pytest.mark.sqlite
+    def test_and_condition_simple(self, sa) -> None:
+        engine = SqlAlchemyExecutionEngine(connection_string="sqlite://")
+        and_condition = AndCondition(
+            conditions=[
+                ComparisonCondition(
+                    column=Column(name="age"), operator=Operator.GREATER_THAN, parameter=18
+                ),
+                ComparisonCondition(
+                    column=Column(name="age"), operator=Operator.LESS_THAN, parameter=65
+                ),
+            ]
+        )
+        result = engine.condition_to_filter_clause(and_condition)
+        compiled = str(result.compile(compile_kwargs={"literal_binds": True}))
+        assert compiled == "age > 18 AND age < 65"
+
+    @pytest.mark.sqlite
+    def test_or_condition_simple(self, sa) -> None:
+        engine = SqlAlchemyExecutionEngine(connection_string="sqlite://")
+        or_condition = OrCondition(
+            conditions=[
+                ComparisonCondition(
+                    column=Column(name="status"), operator=Operator.EQUAL, parameter="active"
+                ),
+                ComparisonCondition(
+                    column=Column(name="status"), operator=Operator.EQUAL, parameter="pending"
+                ),
+            ]
+        )
+        result = engine.condition_to_filter_clause(or_condition)
+        compiled = str(result.compile(compile_kwargs={"literal_binds": True}))
+        assert compiled == "status = 'active' OR status = 'pending'"
+
+    @pytest.mark.sqlite
+    def test_nested_conditions(self, sa) -> None:
+        engine = SqlAlchemyExecutionEngine(connection_string="sqlite://")
+        or_condition = OrCondition(
+            conditions=[
+                AndCondition(
+                    conditions=[
+                        ComparisonCondition(
+                            column=Column(name="age"),
+                            operator=Operator.GREATER_THAN_OR_EQUAL,
+                            parameter=18,
+                        ),
+                        ComparisonCondition(
+                            column=Column(name="age"),
+                            operator=Operator.LESS_THAN_OR_EQUAL,
+                            parameter=65,
+                        ),
+                    ]
+                ),
+                ComparisonCondition(
+                    column=Column(name="status"), operator=Operator.EQUAL, parameter="exempt"
+                ),
+            ]
+        )
+        result = engine.condition_to_filter_clause(or_condition)
+        compiled = str(result.compile(compile_kwargs={"literal_binds": True}))
+        assert compiled == "age >= 18 AND age <= 65 OR status = 'exempt'"
```

Comment:
Nit: these could be parametrized, and I think the simple cases were in the other 2 PRs.